### ISSUE TITLE
Add automatic retries to "Schedule Daily" GHA

### ISSUE THREAD

**Issue by t-will-gillis** [hackforla/website#7120]
### Overview
When reviewing the ["Schedule Daily" workflow logs](https://github.com/hackforla/website/actions/workflows/schedule-daily-1100.yml), it can be seen that many days the workflow fails with the error:
`RequestError [HttpError]: You have exceeded a secondary rate limit. Please wait a few minutes before you try again... ` We need to add code to automatically retry the workflow a set number of times if the workflow does not successfully complete.

### Details
It appears that the `RequestError` occurs in the function `getAllRepos()` in the file `get-project-data.js`. The `RequestError` is due to a secondary rate limit and the message suggests to wait a few minutes before retrying. Therefore we will want to incorporate code that retries the workflow a total of three times: at 2 minutes past the original failure, and if needed after 4 more minutes, and lastly if needed after 8 more minutes. 
### Action Items
- [x] Review one of the recent, failed [workflow logs](https://github.com/hackforla/website/actions/runs/9938251117) and 
- [x] cross-reference with [schedule-daily-1100.yml](https://github.com/hackforla/website/blob/gh-pages/.github/workflows/schedule-daily-1100.yml) and [get-project-data.js](https://github.com/hackforla/website/blob/gh-pages/github-actions/trigger-schedule/github-data/get-project-data.js). 
- [x] Determine the appropriate location for the retry, for example at `getAllRepos()`. *** 
- [x] Add code for automatically retrying if there is a `RequestError`, and set the intervals of 2, then 4, and finally 8 minutes. 
- [x] Test in your own repo that the retry works as intended. Possible test: recreate a `RequestError` in "Schedule Daily" in your repo to confirm that the retries occur.

*** There may be other appropriate ways to retry the failed workflows, for example: adding a retry step after "Generate github-data.json" in `schedule-daily-1100.yml`. FYI- Do not use a GitHub Marketplace action/automation.
### Resources/Instructions
- Info regarding ["Schedule Daily" and testing](https://github.com/hackforla/website/issues/6643#issuecomment-2132043028)

**Comment by HackforLABot** [comment]
Hi @daras-cu, thank you for taking up this issue! Hfla appreciates you :)

Do let fellow developers know about your:-
i. **Availability:** (When are you available to work on the issue/answer questions other programmers might have about your issue?)
ii. **ETA:** (When do you expect this issue to be completed?)

You're awesome!

P.S. - You may not take up another issue until this issue gets merged (or closed). Thanks again :)

**Comment by daras-cu** [comment]
Availability: Fri afternoon, Sat-Mon all day
ETA: ~~Sunday 8/4 EOD~~ Updated to Friday 8/9 EOD

**Comment by daras-cu** [comment]
Weekly update:

Progress: I have completed work on this issue and opened a PR (#7262). I made requested changes from one reviewer.
Blockers: I am waiting on a second review.
Availability: Saturday-Monday all day, weekday evenings
ETA: Unknown, waiting on review
